contracts/facets/GBMFacet.sol

```diff
@@ -222,21 +252,35 @@ contract GBMFacet is IGBM, IERC1155TokenReceiver, IERC721TokenReceiver, Modifier
     /// @param _tokenKind either bytes4(keccak256("ERC721")) or bytes4(keccak256("ERC1155"))
     /// @param _useInitiator Set to `false` if you want to use the default value registered for the token contract (if wanting to reset to default,
     /// use `true`)
-    /// @param _1155Index Set to 0 if dealing with an ERC-721 or registering new 1155 test. otherwise, set to relevant index you want to reinitialize
-    /// @param _rewrite Set to true if you want to rewrite the data of an existing auction, false otherwise
+    /// @param _index Set to 0 if dealing with a registration. Set to desired auction index if you want to reinitialize
+    /// @param _rewriteBlock Set to 0 if you want to register a new auction, to the blocknumber of a past registered auction if you want to edit it
     function modifyAnAuctionToken(
         address _tokenContract,
         uint256 _tokenId,
         bytes4 _tokenKind,
         bool _useInitiator,
-        uint256 _1155Index,
-        bool _rewrite
+        uint256 _index,
+        uint256 _rewriteBlock
     ) internal {
-        if (!_rewrite) {
-            _1155Index = s.erc1155TokensIndex[_tokenContract][_tokenId]; //_1155Index was 0 if creating new auctions
-            require(s.auctionMapping[_tokenContract][_tokenId][_1155Index] == 0, "The auction aleady exist for the specified token");
-        } else {
-            require(s.auctionMapping[_tokenContract][_tokenId][_1155Index] != 0, "The auction doesn't exist yet for the specified token");
+
+		uint256 _locIndex;
+		
+        if (_rewriteBlock != 0) { //Case where we are rewriting an auction
+            if (_tokenKind == bytes4(keccak256("ERC721"))) {
+                require(s.auction721Mapping[_tokenContract][_tokenId][_index] != 0, "The auction doesn't exist for the specified parameters");
+            } else {
+                require(s.auction1155Mapping[_tokenContract][_tokenId][_index] != 0, "The auction doesn't exist for the specified parameters");
```

Comment:
`s.auction1155Mapping` is used for all other `_tokenKinds` which might not be `bytes4(keccak256('ERC1155')`
..better to have a nother `if`block which specifically takes care of `ERC1155` token kinds..reverts if its not either